docs/modules/develop/pages/guide_accessibility.adoc

```diff
@@ -0,0 +1,236 @@
+
+The automated accessibility auditing tools only cover about 50% of the total accessibility requirements. Although a good starting point, the functionality always needs to be tested these with the real tools and compared to working examples. See the "Testing" section below for more information on how to test your service or the elements you are building with different accessibility tools.
+
+=== Valid HTML
+
+Always ensure the web pages have valid HTML. You can test this by copying at the source of the page and pasting it into the https://validator.w3.org/#validate_by_input[HTML validator tool].
+
+Some general guidelines for making the page content valid HTML:
+
+- The page should always have only one `<main>` element and one only. Avoid using this element in the partials and sub-views if the layout already wraps the page contents into this element.
+- Use the `<section>` element with caution but when you decide to use it, ensure it has an identifying heading element (h2-h6).
+- Use the `<article>` element with caution but when you decide to use it, ensure that you are semantically defining an article level content. For example, a proposal **card** or a blog article **teaser** are not semantically articles that define "A complete, or self-contained, composition in a document, page, application, or site and that is, in principle, independently distributable or reusable, e.g. in syndication." If you are using an `<article>` tag, ensure you are describing the full content of the article and also defining the heading element inside the article (see same issue for `<section>` element).
+- When defining an ID attribute for the elements, ensure the values of those attributes are unique on the page. E.g. when a filtering form is doubled for desktop and mobile, the doubled filtering control inside that form needs to have a unique ID in both of its parent elements. So if you are e.g. repeating the same partial multiple times on the page, make sure any ID attributes it contains are always unique in each instance of the generated HTML.
+
+=== Accessibility tests
+
+Once you are done developing some page or fixing its accessibility issues, you should add the following shared examples for the system specs of that page:
+
+[source,ruby]
+----
+it_behaves_like "accessible page"
+----
+
+This checks if the page passes the technical accessibility evaluation and contains valid HTML. It will also ensure that when other people are implementing changes on the pages covered by these tests, they will likely stay accessible also after the changes. Otherwise accessibility can be easily broken as these kinds of issues can be hard to notice for code reviewers.
```

Comment:
```suggestion
This checks if the page passes the technical accessibility evaluation and contains valid HTML. It will also ensure that when other people are implementing changes on the pages covered by these tests, they will likely stay accessible also after the changes. Otherwise, accessibility can be easily broken as these kinds of issues can be hard to notice for code reviewers.
```